components/Article.jsx

```diff
@@ -0,0 +1,33 @@
+import { useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+import { fetchArticlesByID,} from '../api';
+import ArticleComments from './ArticleComments';
+
+
+const Article = () => {
+    const { articleId } = useParams();
+    const [article, setArticle] = useState(null);
+    useEffect(() => {
+        fetchArticlesByID(articleId).then(({article}) => {
+          setArticle(article);
+        });
+      }, [articleId]);
+
+      if (!article) {
+        return <div>Loading...</div>;
+    }
```

Comment:
I see how this is working, however, as we are in React, it is much more efficient to have a state handle the loading message. You can then conditionally render this in your return below depending on if there is an article or not.